### PR TITLE
Adds delete quest command plus a few guards and tests

### PR DESCRIPTION
**Pull request by Rotorsoft**
## Link to Issue
Closes: #9386

## Description of Changes
- Adds new command
- Adds guards for time periods, and updates existing commands to use new guards
- Updates lifecycle test to include deletes and new guards

## Test Plan
- Unit tested  
## Deployment Plan

1. 

## Other Considerations

-

### CODE REVIEW

File: libs/model/src/middleware/guards.ts [hunk 1]
```diff
@@ -106,3 +107,53 @@ export function mustBeAuthorizedComment(actor: Actor, auth?: AuthContext) {
...
+  // check that end date is after start date by more than minDaysInRange
+  const rangeInDays = end.diff(start, 'days');
+  if (rangeInDays < minDaysInRange)
+    throw new InvalidState(
+      `Start ${start.format('YYYY-MM-DD')} and End ${end.format('YYYY-MM-DD')} dates range must be at least ${minDaysInRange} days apart, but are ${rangeInDays} days apart`,
+      { start_date, end_date },
+    );
+
+  return { start, end };
+}
```
Comment (rbennettcw) [inline libs/model/src/middleware/guards.ts:146]: Isolated unit test would be nice e.g. to check for off-by-one error, but up to you

**Review by rbennettcw — APPROVED** [review]

File: libs/model/src/middleware/guards.ts (hunk 1 above)
Comment (Rotorsoft) [reply libs/model/src/middleware/guards.ts:146]: we can if coverage is not 100%, but I think we'll cover the most important parts with lifecycle tests

File: libs/model/src/middleware/guards.ts [hunk 2]
```diff
@@ -106,3 +107,53 @@ export function mustBeAuthorizedComment(actor: Actor, auth?: AuthContext) {
...
+ * @param minDaysInRange
+ * @returns { start, end } start and end dates as moment objects
+ */
+export function mustBeValidDateRange(
+  start_date: Date,
+  end_date: Date,
+  startDaysInTheFuture = 1,
+  minDaysInRange = 1,
+) {
+  const today = moment(new Date());
```
Comment (timolegros) [inline libs/model/src/middleware/guards.ts:125]: We should probably look at moving away from moment.js given that it is self-declared legacy and in-maintenance mode.
Something like Day.js or date-fns.

**Review by timolegros — APPROVED** [review]

File: libs/model/src/middleware/guards.ts (hunk 2 above)
Comment (Rotorsoft) [reply libs/model/src/middleware/guards.ts:125]: 100% - we need a maintenance ticket here https://github.com/hicommonwealth/commonwealth/issues/5400

**Pull request merged**